### DISCUSSION TITLE
Symmetry Number

### DISCUSSION THREAD

**Discussion by TheophileGaudin** [rdkit/rdkit#5798] [Q&A] [answered]
Hi,

Is there any way to calculate the [Symmetry Number](https://en.wikipedia.org/wiki/Symmetry_number) of molecules with rdkit?

Definition:

> The symmetry number or symmetry order of an object is the number of different but indistinguishable (or equivalent) arrangements (or views) of the object, i.e. the order of its symmetry group

Thanks in advance!

**Comment by bp-kelley** [comment]
CanonicalRankAtoms is what you want

>>> mol = MolFromSmiles('C1NCN1')
>>> list(CanonicalRankAtoms(mol, breakTies=False))
[0,1,0,1]

----
Brian Kelley

**Reply by TheophileGaudin** [reply]
@bp-kelley 
Hi,

Thanks for your reply. I am not sure I understand though. For example for Benzene the symmetry number is 12 as you can rotate benzene by 60° (6 configurations) and you can flip it and rotate it 6 other times when flipped leading to 12 identical configurations. How does the above code snippet lead to getting 12 for benzene?

Sorry if the reply is obvious.

**Comment by bp-kelley** [comment, marked as answer]
I misunderstood your question, this function returns the symmetry order for each atom, not the molecule it self.

It will be easier to use GetSubstructMatches to find what you want:

```
>>> m = Chem.MolFromSmiles("c1ccccc1")
>>> len(m.GetSubstructMatches(m, uniquify=False))
12
>>> m = Chem.MolFromSmiles("c1ccccc1C")
>>> len(m.GetSubstructMatches(m, uniquify=False))
2
```

**Reply by TheophileGaudin** [reply]
Thank you very much, that answers my question!
